app/src/main/java/org/simple/clinic/newentry/country/di/InputFieldsFactoryModule.kt

```diff
@@ -0,0 +1,33 @@
+package org.simple.clinic.newentry.country.di
+
+import dagger.Module
+import dagger.Provides
+import org.simple.clinic.appconfig.Country
+import org.simple.clinic.newentry.country.BangladeshInputFieldsProvider
+import org.simple.clinic.newentry.country.EthiopiaInputFieldsProvider
+import org.simple.clinic.newentry.country.IndiaInputFieldsProvider
+import org.simple.clinic.newentry.country.InputFieldsProvider
+import org.simple.clinic.util.UserClock
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import javax.inject.Named
+
+@Module
+class InputFieldsFactoryModule {
+
+  @Provides
+  fun provideInputFieldsProvider(
+      country: Country,
+      @Named("date_for_user_input") dateTimeFormatter: DateTimeFormatter,
```

Comment:
Since we are touching this `Named` annotation should we consider changing it to `Qualifier`?

Reply:
In the interest of time, I did not do this as part of this feature. Will do it once the PRs are merged in.